src/rf/apps/core/models.py

```diff
@@ -133,6 +133,10 @@ class Meta:
     thumb_large_key = CharField(max_length=255, blank=True, default='',
                                 help_text='S3 key for large thumbnail')
 
+    def has_copied_images(self):
+        image = self.layer_images.first()
```

Comment:
I think this logic should be changed to check all the images which would prevent failures if using mixed sources.